### PR TITLE
Fixes #1684: Support CREATE MATERIALIZED VIEW with AUTO REFRESH

### LINKED ISSUE #1684: Redshift MATERIALIZED VIEW with AUTO REFRESH is not supported
Using the ["AUTO REFRESH" clause for a materialized view in Redshift](https://docs.aws.amazon.com/redshift/latest/dg/materialized-view-create-sql-command.html) is not supported by JSqlParser. 

Trying to parse a statement like this:
`CREATE MATERIALIZED VIEW AUTO REFRESH YES foo AS SELECT * FROM bar`
results in:

```
net.sf.jsqlparser.JSQLParserException: Encountered unexpected token: "CREATE" "CREATE"
    at line 1, column 1.

Was expecting one of:

    "("
    "ALTER"
    "CALL"
    "COMMENT"
    "COMMIT"
    "DECLARE"
    "DELETE"
    "DESCRIBE"
    "DROP"
    "EXEC"
    "EXECUTE"
    "EXPLAIN"
    "GRANT"
    "INSERT"
    "MERGE"
    "PURGE"
    "RENAME"
    "RESET"
    "ROLLBACK"
    "SAVEPOINT"
    "SET"
    "SHOW"
    "TRUNCATE"
    "UPDATE"
    "UPSERT"
    "USE"
    "VALUES"
    "WITH"
    <K_SELECT>

	at app//net.sf.jsqlparser.parser.CCJSqlParserUtil.parseStatement(CCJSqlParserUtil.java:190)
	at app//net.sf.jsqlparser.parser.CCJSqlParserUtil.parse(CCJSqlParserUtil.java:63)
	at app//net.sf.jsqlparser.parser.CCJSqlParserUtil.parse(CCJSqlParserUtil.java:38)
	... 86 more
Caused by: net.sf.jsqlparser.parser.ParseException: Encountered unexpected token: "CREATE" "CREATE"
    at line 1, column 1.
```

With the `AUTO REFRESH YES` clause removed the statement is parsed just fine.

### PR DESCRIPTION
**Pull request by zaza**
Support parsing create view statements in Redshift with AUTO REFRESH option.

### CODE REVIEW

**Comment by wumpz** [comment]
please fix those branch conflicts first

**Comment by zaza** [comment]
> please fix those branch conflicts first

Done with Github UI so fingers crossed.

**Comment by wumpz** [comment]
You still have some PMD violations there. Look at the failing tests at GitHub.

**Comment by zaza** [comment]
@wumpz can please help me understand how a failure in `SpecialOracleTest.testAllSqlsParseDeparse` for file `query_factoring01.sql` is related to this change? I'm confused.

**Comment by wumpz** [comment]
@zaza In the log file is written: 

```
INFO: tested 276 files. got 195 correct parse results, expected 196
```

This oracle test scans through a list of over 200 sqls. It expects 196 parse successes. Obviously your change somehow degraded the grammar so one sql could not be parsed anymore.

**Comment by zaza** [comment]
Thanks @wumpz, but I got that part. I just don't understand how my change (in how CREATE VIEW is parsed/deparsed) could possibly broke the test. Like I said in my previous comment, the affected test file is `query_factoring01.sql` and the error message is `--@FAILURE: Encountered unexpected token: "," "," recorded first on Jan 29, 2023, 10:56:13 PM` (just ran it again). What puzzles me is that this is not a CREATE VIEW statement. I'm sorry, I'm new to this project and I'm probably missing something obvious.

File: src/main/java/net/sf/jsqlparser/parser/ParserKeywordsUtils.java [hunk 1]
```diff
@@ -182,6 +182,11 @@ public static List<String> getReservedKeywords(int restriction) {
 
                 , { "NEXTVAL", RESTRICTED_JSQLPARSER }
 
+                , { "AUTO", RESTRICTED_JSQLPARSER }
```
Comment (manticore-projects) [inline src/main/java/net/sf/jsqlparser/parser/ParserKeywordsUtils.java:185]: Are those really restricted Keywords which must not been used?
Or are those just keywords? Why exactly have those Tokens been added?

**Comment by manticore-projects** [comment]
@zaza: I have merged your changes into PR #1722 and fixed it by remove the unnecessary `Reserved Keywords`, you have introduced. You can check the new syntax here https://manticore-projects.github.io/JSqlParser/syntax.html#createview

In general, keywords must only be reserved, when absolutely needed to keep the parser working (and not in this case).

Thank you for your contribution! Please consider to close this PR.

**Comment by zaza** [comment]
Thanks @manticore-projects , closing the PR as requested.

**Pull request closed without merging**